float64 pixel_x
float64 pixel_y
float64 depth